Framework/Algorithms/src/ConvertSpectrumAxis.cpp

```diff
@@ -95,15 +96,16 @@ void ConvertSpectrumAxis::exec() {
       emode = 2;
     const double delta = 0.0;
     double efixed;
+    auto spectrumInfo = inputWS->spectrumInfo();
     for (size_t i = 0; i < nHist; i++) {
       std::vector<double> xval{inputWS->x(i).front(), inputWS->x(i).back()};
-      IDetector_const_sptr detector = inputWS->getDetector(i);
       double twoTheta, l1val, l2;
-      if (!detector->isMonitor()) {
-        twoTheta = inputWS->detectorTwoTheta(*detector);
-        l2 = detector->getDistance(*sample);
+      if (!spectrumInfo.isMonitor(i)) {
+        twoTheta = spectrumInfo.twoTheta(i);
+        l2 = spectrumInfo.l2(i);
         l1val = l1;
-        efixed = getEfixed(detector, inputWS, emode); // get efixed
+        efixed =
+            getEfixed(spectrumInfo.detector(i), inputWS, emode); // get efixed
```

Comment:
I would agree - it is pretty widely used but the code to get e-fixed can be messy. I think if we make this part of `SpectrumInfo` it will make the rollout easier.